Code/Tools/AssetProcessor/native/unittests/AssetProcessingStateDataUnitTests.cpp

```diff
@@ -595,123 +172,793 @@ namespace AssetProcessor
+                }
+            }
+            return false;
+        };
+
+        auto ProductDependenciesContainProductID = [](const ProductDependencyDatabaseEntryContainer& productDependencies, AZ::s64 productId) -> bool
+        {
+            for (const auto& productDependency : productDependencies)
+            {
+                if (productDependency.m_productPK == productId)
+                {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        auto ProductDependenciesContainDependencySoureGuid = [](const ProductDependencyDatabaseEntryContainer& productDependencies, AZ::Uuid dependencySourceGuid) -> bool
+        {
+            for (const auto& productDependency : productDependencies)
+            {
+                if (productDependency.m_dependencySourceGuid == dependencySourceGuid)
+                {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        auto ProductDependenciesContainDependencySubID = [](const ProductDependencyDatabaseEntryContainer& productDependencies, AZ::u32 dependencySubID) -> bool
+        {
+            for (const auto& productDependency : productDependencies)
+            {
+                if (productDependency.m_dependencySubID == dependencySubID)
+                {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        auto ProductDependenciesContainDependencyFlags = [](const ProductDependencyDatabaseEntryContainer& productDependencies, AZStd::bitset<64> dependencyFlags) -> bool
+        {
+            for (const auto& productDependency : productDependencies)
+            {
+                if (productDependency.m_dependencyFlags == dependencyFlags)
+                {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        auto ProductsContainProductID = [](const ProductDatabaseEntryContainer& products, AZ::s64 productId) -> bool
+        {
+            for (const auto& product : products)
+            {
+                if (product.m_productID == productId)
+                {
+                    return true;
+                }
+            }
+            return false;
+        };
+    }
+
+    class AssetProcessingStateDataUnitTest
+        : public UnitTest::AssetProcessorUnitTestBase
+    {
+    protected:
+        void SetUp() override
+        {
+            UnitTest::AssetProcessorUnitTestBase::SetUp();
+
+            EXPECT_TRUE(m_connection.DataExists());
+            EXPECT_TRUE(m_connection.OpenDatabase());
+        }
+
+        void TearDown() override
+        {
+            m_connection.CloseDatabase();
+
+            UnitTest::AssetProcessorUnitTestBase::TearDown();
+        }
+
+        void AddDefaultScanFolder()
+        {
+            m_defaultScanFolder = ScanFolderDatabaseEntry("c:/O3DE/dev", "dev", "rootportkey");
```

Comment:
It's probably worth adding a comment that this isn't a real directory. I was about to ask if this would work on Linux, but I think this just creates an entry in the database, so any string would work.